### PR TITLE
More metrics

### PR DESCRIPTION
**Pull request by netoneko**
Related to #266

Updating default configuration in 83a9bf6 was blessed by @talkol

### CODE REVIEW

File: services/consensuscontext/create_block.go [hunk 1]
```diff
@@ -9,6 +9,9 @@ import (
 )
 
 func (s *service) createTransactionsBlock(blockHeight primitives.BlockHeight, prevBlockHash primitives.Sha256) (*protocol.TransactionsBlockContainer, error) {
+	meter := s.reporting.Meter("tx-block-creation")
```
Comment (electricmonk) [inline services/consensuscontext/create_block.go]: could be a one-liner

File: services/publicapi/service.go [hunk 2]
```diff
@@ -65,6 +65,9 @@ func (s *service) SendTransaction(input *services.SendTransactionInput) (*servic
 	s.reporting.Info("transaction received via public api", log.String("flow", "checkpoint"), log.Stringable("txHash", txHash))
 	defer s.reporting.Info("transaction status returned by public api", log.String("flow", "checkpoint"), log.Stringable("txHash", txHash))
 
+	meter := s.reporting.Meter("tx-processing-time", log.Stringable("txHash", txHash))
+	defer meter.Done()
```
Comment (electricmonk) [inline services/publicapi/service.go]: I'm not sure this is good enough. Do we want to collect different metrics, or to qualify at least, the difference between a tx that has finished successfully, a tx that timed out (i.e. result did not return within timeout interval) and a tx that failed? if so, how would we do that?

File: services/publicapi/test/public_api_test.go [hunk 3]
```diff
@@ -86,6 +86,45 @@ func TestSendTransaction_BlocksUntilTransactionCompletes(t *testing.T) {
 	})
 }
 
+func TestSendTransaction_TimesOut(t *testing.T) {
```
Comment (electricmonk) [inline services/publicapi/test/public_api_test.go]: why did you add this test?

Comment (netoneko) [reply services/publicapi/test/public_api_test.go]: There wasn't one and I needed to check if the timeout works as expected.

File: services/publicapi/service.go (hunk 2 above)
Comment (netoneko) [reply services/publicapi/service.go]: This is a good point

Comment (electricmonk) [reply services/publicapi/service.go]: we can move forward with this, but document the downsides fo this approach in a new issue

File: services/consensusalgo/benchmarkconsensus/leader.go [hunk 4]
```diff
@@ -26,6 +26,8 @@ func (s *service) leaderConsensusRoundRunLoop(ctx context.Context) {
 
 	s.lastCommittedBlock = s.leaderGenerateGenesisBlock()
 	for {
+		meter := s.logger.Meter("consensus-round-tick")
```
Comment (electricmonk) [inline services/consensusalgo/benchmarkconsensus/leader.go]: here as well, a better approach would be to measure distinct branches of behavior: one for successful round, one for timeout and one for error. The API for `Meter` might need to change - can you think of a better approach?

File: services/publicapi/test/public_api_test.go [hunk 5]
```diff
@@ -110,13 +112,54 @@ func newPublicApiConfig(txTimeout time.Duration) publicapi.Config {
...
+		vmMock := &services.MockVirtualMachine{}
+
+		timeoutDuration := 1 * time.Millisecond
+
+		cfg := newPublicApiConfig(timeoutDuration)
+		papi := publicapi.NewPublicApi(ctx, cfg, txpMock, vmMock, logger)
+
+		blockTime := primitives.TimestampNano(time.Now().Nanosecond())
+		txb := builders.Transaction().Builder()
+		txpMock.When("AddNewTransaction", mock.Any).Times(1).
```
Comment (electricmonk) [inline services/publicapi/test/public_api_test.go]: take a look at master for this code, @noambergIL introduced a test harness, make sure you adapt to it

File: services/publicapi/test/public_api_test.go [hunk 6]
```diff
@@ -110,13 +112,54 @@ func newPublicApiConfig(txTimeout time.Duration) publicapi.Config {
...
+		tx, err := papi.SendTransaction(&services.SendTransactionInput{
+			ClientRequest: (&client.SendTransactionRequestBuilder{
+				SignedTransaction: txb,
+			}).Build(),
+		})
+
+		txHash := digest.CalcTxHash(txb.Build().Transaction())
+
+		require.EqualError(t, err, fmt.Sprintf("timed out waiting for transaction result %s", txHash.String()))
+		require.WithinDuration(t, time.Now(), start, timeoutDuration*2, "timeout duration exceeded")
```
Comment (electricmonk) [inline services/publicapi/test/public_api_test.go]: very nice way to enforce timeouts! can you share this approach on Slack? I for one didn't think of this

File: test/e2e/harness.go [hunk 7]
```diff
@@ -65,7 +65,13 @@ func newHarness() *harness {
 			gossipPeers[publicKey.KeyForMap()] = config.NewHardCodedGossipPeer(uint16(firstRandomPort+i), "127.0.0.1")
 		}
 
-		logger := log.GetLogger().WithOutput(log.NewOutput(os.Stdout).WithFormatter(log.NewHumanReadableFormatter()))
+		os.MkdirAll(config.GetProjectSourceRootPath()+"/logs", 0755)
+
+		logger := log.GetLogger().WithTags(
+			log.String("_test", "e2e"),
+			log.String("_branch", os.Getenv("GIT_BRANCH")),
```
Comment (electricmonk) [inline test/e2e/harness.go]: these two tags can be added automatically during enrichment, where you add the caller info

Comment (netoneko) [reply test/e2e/harness.go]: I prefer this code to be explicitly outside for now. If we see that we use it in multiple places I will move it.

File: services/publicapi/test/public_api_test.go (hunk 6 above)
Comment (netoneko) [reply services/publicapi/test/public_api_test.go]: Done.

File: services/consensusalgo/benchmarkconsensus/leader.go (hunk 4 above)
Comment (netoneko) [reply services/consensusalgo/benchmarkconsensus/leader.go]: I think it makes sense, we should rethink the API.

File: services/publicapi/test/public_api_test.go (hunk 5 above)
Comment (netoneko) [reply services/publicapi/test/public_api_test.go]: Done.

**Comment by netoneko** [comment]
test
